lua/neo-tree/sources/filesystem/commands.lua

```diff
@@ -222,14 +222,13 @@ M.rename = function(state)
 end
 
 M.set_root = function(state)
-  local tree = state.tree
-  local node = tree:get_node()
-  if node.type == "directory" then
-    if state.search_pattern then
-      fs.reset_search(state, false)
-    end
-    fs._navigate_internal(state, node.id, nil, nil, false)
+  if state.search_pattern then
+    fs.reset_search(state, false)
   end
+
+  local node = state.tree:get_node()
+  local id = node.type == "directory" and node.id or node._parent_id
```

Comment:
No, access to `node.id` and `node._parent_id` is not allowed.

Use the get methods listed in the doc.
https://github.com/MunifTanjim/nui.nvim/tree/main/lua/nui/tree#nuitreenode